vmngclient/session.py

```diff
@@ -462,18 +469,13 @@ def __init__(
         self.subdomain = subdomain
         super().__init__(url, port, username, password, timeout)
 
-    def login(self) -> None:
-        """Logs in to vManage API as Provider using username/password and switches to Tenant."""
-        super().login()
-        self.__switch_to_tenant()
-
-    def __get_tenant_id(self) -> str:
+    def __get_tenant_id(self, subdomain: str) -> str:
         """Gets tenant UUID for tenant subdomain.
         Returns:
             tenant UUID
         """
         tenants = self.get_data('/dataservice/tenant')
-        tenant_ids = [tenant.get('tenantId') for tenant in tenants if tenant['subDomain'] == self.subdomain]
+        tenant_ids = [tenant.get('tenantId') for tenant in tenants if tenant['subDomain'] == subdomain]
         assert len(tenant_ids) > 0, f"Tenant not found for subdomain: {self.subdomain}"
```

Comment:
this assert message is wrong as we check subdomain and message prints self.subdomain which could be different

Reply:
fixed, all use `self.subdomain`